core/storage/statistics/gae_models.py

```diff
@@ -1134,6 +1136,224 @@ def delete_playthroughs_multi(cls, playthrough_ids):
+            feconf.ENTITY_TYPE_EXPLORATION, state_reference)
+        learner_answer_info_dict_size = cls._get_learner_answer_info_dict_size(
+            new_learner_answer_info)
+        if model_instance:
+            if (model_instance.accumulated_answer_info_json_size_bytes) + (
+                    learner_answer_info_dict_size) <= (
+                        cls._MAX_LEARNER_ANSWER_INFO_LIST_BYTE_SIZE):
+                model_instance.learner_answer_info_list.append(
+                    new_learner_answer_info)
+                model_instance.accumulated_answer_info_json_size_bytes += (
+                    learner_answer_info_dict_size)
+                model_instance.put()
+        else:
+            cls.create_model_instance_for_exploration(
+                exploration_id, state_name, interaction_id,
+                new_learner_answer_info)
+
+    @classmethod
+    def insert_learner_answer_info_for_question(
+            cls, question_id, interaction_id, new_learner_answer_info):
+        """Insert a new learner answer info in the learner answer info list in
+        the existing model instance of the state, if the model instance for the
+        state does not exist then the model instance is created.
+
+        Args:
+            question_id: str. The ID of the question.
+            interaction_id: str. The ID of the interaction.
+            new_learner_answer_info: dict. The dict representation of
+                LearnerAnswerInfo class in stats_domain.
+        """
+        state_reference = cls.get_state_reference_for_question(question_id)
+        model_instance = cls.get_model_instance(
+            feconf.ENTITY_TYPE_QUESTION, state_reference)
+        learner_answer_info_dict_size = cls._get_learner_answer_info_dict_size(
+            new_learner_answer_info)
+        if model_instance:
+            if (model_instance.accumulated_answer_info_json_size_bytes) + (
+                    learner_answer_info_dict_size) <= (
+                        cls._MAX_LEARNER_ANSWER_INFO_LIST_BYTE_SIZE):
+                model_instance.learner_answer_info_list.append(
+                    new_learner_answer_info)
+                model_instance.accumulated_answer_info_json_size_bytes += (
+                    learner_answer_info_dict_size)
+            model_instance.put()
+        else:
+            cls.create_model_instance_for_question(
+                question_id, interaction_id, new_learner_answer_info)
+
+    @classmethod
+    def create_model_instance_for_exploration(
+            cls, exp_id, state_name, interaction_id, learner_answer_info):
+        """Creates a new LearnerAnswerDetailsModel for the entity type
+        exploration and then writes it to the datastore.
+
+        Args:
+            exp_id: str. ID of the exploration.
+            state_name: str. The name of the state.
+            interaction_id: str. The id of the interaction.
+            learner_answer_info: dict. The dict representation of
```

Comment:
Here and elsewhere, perhaps you should call it learner_answer_info_dict to distinguish it from the domain object.

Reply:
Done